From df849f8ea8750e934a7a2c9ea2d5628b0c209056 Mon Sep 17 00:00:00 2001
From: Mohammed Hawari <mohammed@hawari.fr>
Date: Tue, 19 Jan 2021 18:19:45 +0100
Subject: [PATCH] rdma: fix pkg_config file

When building DPDK with rdma linkage, this patch avoids linking against
useless verb providers. It also hard-codes the library directory to lib
to fix CentOS behavior.

Change-Id: I3acd94adf1b7e59e023346b3c254bd4bba6157df
Type: fix
Signed-off-by: Mohammed Hawari <mohammed@hawari.fr>
---
 build/external/packages/rdma-core.mk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/build/external/packages/rdma-core.mk b/build/external/packages/rdma-core.mk
index 65c9c8aa6..960a2ffa1 100644
--- a/build/external/packages/rdma-core.mk
+++ b/build/external/packages/rdma-core.mk
@@ -39,6 +39,7 @@ define  rdma-core_config_cmds
 	  $(CMAKE) -G Ninja $(rdma-core_src_dir) \
 	    -DENABLE_STATIC=1 -DENABLE_RESOLVE_NEIGH=0 -DNO_PYVERBS=1 -DENABLE_VALGRIND=0\
 	    -DCMAKE_BUILD_TYPE=$(RDMA_BUILD_TYPE) -DCMAKE_INSTALL_PREFIX=$(rdma-core_install_dir) \
+	    -DCMAKE_INSTALL_LIBDIR=lib \
 	    -DCMAKE_C_FLAGS='-fPIC -fvisibility=hidden' > $(rdma-core_config_log)
 endef
 
@@ -52,6 +53,8 @@ define  rdma-core_install_cmds
 	find $(rdma-core_install_dir) -name '*.a' -exec mv -v {} $(rdma-core_install_dir)/lib \; >> $(rdma-core_install_log)
 	rmdir -v $(rdma-core_install_dir)/util $(rdma-core_install_dir)/lib/statics >> $(rdma-core_install_log)
 	sed '/Libs.private:/ s/$$/ -lrdma_util -lccan/' -i $(rdma-core_install_dir)/lib/pkgconfig/libibverbs.pc
+	sed '/Libs.private:/ s/ \S*\(rdmav25\)\S*//g'   -i $(rdma-core_install_dir)/lib/pkgconfig/libibverbs.pc
+	sed '/Libs.private:/ s/-lefa//g'                -i $(rdma-core_install_dir)/lib/pkgconfig/libibverbs.pc
 endef
 
 $(eval $(call package,rdma-core))
-- 
2.17.1

